cpp/include/experimental/graph_view.hpp

```diff
@@ -428,6 +444,33 @@ class graph_view_t<vertex_t,
 
   bool is_hypergraph_partitioned() const { return partition_.is_hypergraph_partitioned(); }
 
+  // FIXME: this function is not part of the public stable API.This function is mainly for pattern
+  // accelerator implementation. This function is currently public to support the legacy
+  // implementations directly accessing CSR/CSC data, but this function will eventually become
+  // private or even disappear if we switch to CSR + DCSR (or CSC + DCSC).
+  edge_t const* offsets() const
+  {
+    return offsets(0);
+  }
+
+  // FIXME: this function is not part of the public stable API.This function is mainly for pattern
+  // accelerator implementation. This function is currently public to support the legacy
+  // implementations directly accessing CSR/CSC data, but this function will eventually become
+  // private or even disappear if we switch to CSR + DCSR (or CSC + DCSC).
+  vertex_t const* indices() const
+  {
+    return indices(0);
+  }
+
+  // FIXME: this function is not part of the public stable API.This function is mainly for pattern
+  // accelerator implementation. This function is currently public to support the legacy
+  // implementations directly accessing CSR/CSC data, but this function will eventually become
+  // private or even disappear if we switch to CSR + DCSR (or CSC + DCSC).
+  weight_t const* weights() const
+  {
+    return weights(0);
+  }
+
```

Comment:
Better delete this. graph_view's offsets, indices, and weights() are for legacy code and will be deleted.

Use

matrix_partition_device_t<GraphViewType> matrix_partition(graph_view, 0);

This class can be passed to device by value, and has

```
  __device__ thrust::tuple<vertex_t const*, weight_t const*, edge_t> get_local_edges(
    vertex_t major_offset) const noexcept
  {
    auto edge_offset  = *(offsets_ + major_offset);
    auto local_degree = *(offsets_ + (major_offset + 1)) - edge_offset;
    auto indices      = indices_ + edge_offset;
    auto weights      = weights_ != nullptr ? weights_ + edge_offset : nullptr;
    return thrust::make_tuple(indices, weights, local_degree);
  }
```

Reply:
At this point, I think deferring this to 0.17 is wise.